csharp-mode-tests.el

```diff
@@ -165,84 +165,49 @@
            (equal expected (match-string 1)))))
       (kill-buffer buffer))))
 
-(ert-deftest imenu-parsing-supports-default-values ()
-  (dolist (test-case
-           '(;; should support bools
-             ("(bool a, bool b = true)"                  "(bool, bool)")
-             ("(bool a=true, bool b)"                    "(bool, bool)")
-             ;; should support strings
-             ("(string a, string b = \"quoted string\")" "(string, string)")
-             ("(string a = \"quoted string\", string b)" "(string, string)")
-             ;; should support chars
-             ("(char a, char b = 'b')"                   "(char, char)")
-             ("(char a = 'a', char b)"                   "(char, char)")
-             ;; should support self-object-access
-             ("(object o = Const)"                       "(object)")
-             ;; should support other-object-access
-             ("(object o = ConstObject.Const)"           "(object)")
-             ))
-    (let* ((test-value     (car test-case))
-           (expected-value (cadr test-case))
-           (result         (csharp--imenu-remove-param-names-from-paramlist test-value)))
-      (should (equal expected-value result)))))
-
 (defmacro def-imenutest (testname filename index &rest body)
   `(ert-deftest ,testname ()
      (let* ((find-file-hook nil) ;; avoid vc-mode file-hooks when opening!
             (buffer         (find-file-read-only ,filename))
-            (,index         (csharp--imenu-create-index-helper nil "" t t)) ;; same line as in `csharp-imenu-create-index'.
-            )
+            (,index         (csharp--imenu-create-index-function)))
        ,@body
        (kill-buffer buffer))))
 
+(defun imenu-get-item (index haystack)
+  (setq result nil)
```

Comment:
You'll want to use `(let (result) ...)` or `(let ((result nil)) ...)` and modify that local binding with `setq`, otherwise result will be defined as global variable (yay, dynamic binding).